server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -160,7 +174,16 @@ public RemoteRestoreResult restore(
             }
         }
         validate(currentState, indexMetadataMap, restoreClusterUUID, restoreAllShards);
-        return executeRestore(currentState, indexMetadataMap, restoreAllShards);
+        RemoteRestoreResult remoteRestoreResult = executeRestore(currentState, indexMetadataMap, restoreAllShards);
```

Comment:
Do we need to check version before we can restore?

Reply:
@Bukhtawar are you referring to the Metadata.version here or the manifest CODEC version we add in the filepath and header?